plugins/auth-backend/src/providers/azure-easyauth/provider.ts

```diff
@@ -0,0 +1,202 @@
+      accessToken: accessToken,
+    };
+  }
+
+  private idTokenToProfile(idToken: string) {
+    const claims = decodeJwt(idToken);
+
+    if (claims.ver !== '2.0') {
+      throw new Error('id_token is not version 2.0 ');
+    }
+
+    return {
+      id: claims.oid,
+      displayName: claims.name,
+      provider: 'easyauth',
+      emails: [{ value: claims.email }],
+      username: claims.preferred_username,
+    } as Profile;
+  }
+
+  private async handleResult(
+    result: EasyAuthResult,
+  ): Promise<EasyAuthResponse> {
+    const { profile } = await this.authHandler(result, this.resolverContext);
+
+    const backstageIdentity = await this.signInResolver(
+      {
+        result,
+        profile,
+      },
+      this.resolverContext,
+    );
+
+    return {
+      providerInfo: {
+        accessToken: result.accessToken,
+      },
+      backstageIdentity: prepareBackstageIdentityResponse(backstageIdentity),
+      profile,
+    };
+  }
+}
+
+/**
+ * Auth provider integration for Azure EasyAuth
+ *
+ * @public
+ */
+export const easyAuth = createAuthProviderIntegration({
+  create(options?: {
+    /**
+     * The profile transformation function used to verify and convert the auth response
+     * into the profile that will be presented to the user.
+     */
+    authHandler?: AuthHandler<EasyAuthResult>;
+
+    /**
+     * Configure sign-in for this provider, without it the provider can not be used to sign users in.
+     */
+    signIn: {
+      /**
+       * Maps an auth result to a Backstage identity for the user.
+       */
+      resolver: SignInResolver<EasyAuthResult>;
+    };
+  }) {
+    return ({ resolverContext }) => {
+      validateAppServiceConfiguration(process.env);
+
+      if (options?.signIn.resolver === undefined) {
+        throw new Error(
+          'SignInResolver is required to use this authentication provider',
+        );
+      }
+
+      const authHandler =
+        options.authHandler ??
+        (async ({ fullProfile }) => ({
+          profile: makeProfileInfo(fullProfile),
+        }));
+
+      return new EasyAuthAuthProvider({
+        signInResolver: options.signIn.resolver,
+        authHandler,
+        resolverContext,
+      });
+    };
+  },
+});
+
+function validateAppServiceConfiguration(env: NodeJS.ProcessEnv) {
+  // Based on https://github.com/AzureAD/microsoft-identity-web/blob/f7403779d1a91f4a3fec0ed0993bd82f50f299e1/src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationInformation.cs#L38-L59
+  //
+  // It's critical to validate we're really running in a correctly configured Azure App Services,
+  // As we rely on App Services to manage & validate the ID and Access Token headers
+  // Without that, this users can be trivially impersonated.
+  if (env.WEBSITE_SKU === undefined) {
+    throw new Error('Backstage is not running on Azure App Services');
+  }
+  if (!isEqualCaseInsensitive(env.WEBSITE_AUTH_ENABLED, 'True')) {
```

Comment:
I think we can just simplify some of this to be `env.WEBSITE_AUTH_ENABLED?.toLowerCase() === 'true'` instead of using `localCompare`?